src/components/Modal/ModalProvider.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import createReactContext from 'create-react-context';
+
+import Modal, { TRANSITION_DURATION } from './Modal';
+import { childrenPropType } from '../../util/shared-prop-types';
+
+const {
+  Provider: ContextProvider,
+  Consumer: ModalConsumer
+} = createReactContext({
+  setModal: () => {},
+  getModal: () => {}
+});
+
+export { ModalConsumer };
+
+export class ModalProvider extends Component {
+  static propTypes = {
+    children: childrenPropType.isRequired
+  };
+
+  state = {
+    modal: null,
+    isOpen: false
+  };
```

Comment:
Is this a new syntax to set the initial state?

Reply:
Yeah, this is [property initializer syntax](https://www.fullstackreact.com/articles/use-property-initializers-for-cleaner-react-components/). I believe you've used this in a bunch of places yourself already, no? I know I have. Works just like when declaring a method on a class component.

Just one of those future JS features that got picked up early by the React community.

Reply:
I had seen it before but wasn't sure how/when to use it. The article you linked clears it up. Thanks! :)